catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java

```diff
@@ -411,17 +411,20 @@ public void run() {
                 }
 
                 if (startDate != null) {
-                    if (startDate.getTime() > lastExecutionTime) {
-                        shouldRun = false;
+                    long now = System.currentTimeMillis();
+                    if (startDate.getTime() > now) {
+                        long waitToStart = startDate.getTime() - now;
+                        long waitSecs = TimeUnit.MILLISECONDS.toSeconds(waitToStart);
+                        LOGGER.debug("Start time for subscription is in the future, waiting {} seconds", waitSecs);
+
+                        try {
```

Comment:
Is there a limit on the thread pool (for the rather unlikely case we get a ton of future standing query requests)?